src/Common/src/CoreLib/System/Collections/Concurrent/ConcurrentQueueSegment.cs

```diff
@@ -138,9 +138,10 @@ internal static int RoundUpToPowerOf2(int i)
                 // Get the head at which to try to dequeue.
                 int currentHead = Volatile.Read(ref _headAndTail.Head);
                 int slotsIndex = currentHead & _slotsMask;
+                ref Slot slot = ref slots[slotsIndex];
```

Comment:
FWIW, I'd previously tried this use of ref and didn't see any measurable benefits, but you're off course welcome to try again; things may have changed in the interim.

However, ConcurrentQueue itself is built in the coreclr repo, so any changes to this Segment here won't actually be observable as part of CQ benchmarking. It'd be better to experiment in the coreclr repo. It's also not really worth a PR until you have some measurements to justify the changes.

Thanks!

Reply:
Closing it then. Will raise when a potential gain is observed. Thank you @stephentoub